CHANGELOG.md

```diff
@@ -19,6 +19,19 @@
 - Deprecate `componenterror` package, move everything to `component` (#5383)
 - `pcommon.Value.NewValueBytes` is deprecated in favor of `Value.NewValueMBytes` in preparation of migration to 
   immutable slices (#5367)
+- Move `config.Map` to its own package `confmap` which does not depend on any component concept (#5237)
+  - `config.Map` -> `confmap.ConfMap`
```

Comment:
Should this be `confmap.Map`? Seems a bit easier to use, not a strong opinion here

Reply:
or `confmap.Conf`?

Reply:
`confmap.Conf`

Reply:
Last idea is `conf.Map`? Seems that I have 2 votes for `confmap.Conf` so will implement all changes for that proposal.

Reply:
I like `confmap` more because `conf` can be confused with `config`

Reply:
Done.